inc/components/class-menu.php

```diff
@@ -37,6 +37,18 @@ public function default_config() {
 	 * @return Menu An instance of the Menu class.
 	 */
 	public function parse_wp_menu_by_location( string $menu_location ) {
+
+		// Get menu locations.
+		$locations = get_nav_menu_locations();
+
+		// Get object id by location.
+		$menu_term = wp_get_nav_menu_object( $locations[ $menu_location ] ?? null );
+
+		// If valid menu term.
+		if ( $menu_term instanceof \WP_Term ) {
+			$this->parse_wp_menu( $menu_term->slug );
```

Comment:
Just pas the term object vs the slug. Passing the slug causes extra work, as WordPress has to then lookup the term again.